File: 806-number-of-lines-to-write-string.java
class Solution {
    public int[] numberOfLines(int[] widths, String s) {
        int lines = 1, width = 0;
        for(char ch : s.toCharArray()){
            int pixels = widths[ch - 'a'];
            if((width+ pixels)>100){
                lines++;
                width = 0;
            }
            width += pixels;
        }
        return new int[]{lines,width};
    }
}

File: 806-number-of-lines-to-write-string.py
class Solution:
    def numberOfLines(self, widths: List[int], s: str) -> List[int]:
        cnt, pixels =1, 0
        for ch in s:
            val = widths[ord(ch)-ord('a')]
            pixels += val
            if pixels > 100:
                cnt +=1
                pixels = val
        return [cnt,pixels]
